iminuit/_libiminuit.pyx

```diff
@@ -328,25 +372,81 @@ cdef class Minuit:
         self.throw_nan = throw_nan
 
         self.parameters = args
-        self.args = tuple(self.initialvalue[k] for k in args)
-        self.values = {k: self.initialvalue[k] for k in args}
-        self.errors = {k: self.initialerror[k] for k in args}
+        self.args = tuple(self.initial_value[k] for k in args)
+        self.values = {k: self.initial_value[k] for k in args}
+        self.errors = {k: self.initial_error[k] for k in args}
         self.covariance = None
         self.fval = 0.
         self.ncalls = 0
         self.edm = 1.
         self.merrors = {}
         self.gcc = None
-        if pedantic: self.pedantic(kwds)
 
         self.fitarg = {}
-        self.fitarg.update(self.initialvalue)
-        self.fitarg.update({'error_' + k: v for k, v in self.initialerror.items()})
-        self.fitarg.update({'limit_' + k: v for k, v in self.initiallimit.items()})
-        self.fitarg.update({'fix_' + k: v for k, v in self.initialfix.items()})
+        self.fitarg.update(self.initial_value)
+        self.fitarg.update({'error_' + k: v for k, v in self.initial_error.items()})
+        self.fitarg.update({'limit_' + k: v for k, v in self.initial_limit.items()})
+        self.fitarg.update({'fix_' + k: v for k, v in self.initial_fix.items()})
 
         self.merrors_struct = {}
 
+
+    @classmethod
+    def from_array_func(cls, fcn, start, error=None, limit=None, fix=None,
+                        name=None, **kwds):
+        """
+        Construct minuit object from given *fcn* and start sequence.
+
+        This is an alternative named constructor for the minuit object. It is
+        more convenient to use for functions that accept a numpy array.
+
+        **Arguments:**
+
+            **fcn**: The function to be optimized. Must accept a single
+            parameter that is a numpy array.
+
+                def func(x): ...
+
+            **start**: Sequence of numbers. Starting point for the
+            minimization.
+
+        **Keyword arguments:**
+
+            **error**: Optional sequence of numbers. Initial step sizes.
+
+            **limit**: Optional sequence of limits that restrict the range in
+            which a parameter is varied by minuit. Limits can be set in
+            several ways. With inf = float("infinity") we get:
+
+            - No limit: None, (-inf, inf), (None, None)
+
+            - Lower limit: (x, None), (x, inf) [replace x with a number]
+
+            - Upper limit: (None, x), (-inf, x) [replace x with a number]
+
+            **fix**: Optional sequence of boolean values. Whether to fix a
+            parameter to the starting value.
+
+            **name**: Optional sequence of parameter names. If names are not
+            specified, the parameters are called x0, ..., xN.
+
+            All other keywords are forwarded to :class:`Minuit`, see
+            its documentation.
```

Comment:
Working simple example (probably in the high-level docs and only linked to from here) is gold for users.

Reply:
Right, but let's work on the docs in a separate PR, since there is more to do than adding an example for this.